processing/src/main/java/org/apache/carbondata/processing/util/CarbonLoaderUtil.java

```diff
@@ -323,10 +320,6 @@ public static boolean recordNewLoadMetadata(LoadMetadataDetails newMetaEntry,
             for (LoadMetadataDetails entry : listOfLoadFolderDetails) {
```

Comment:
Let's discuss insertOverwrite behavior.
Should we delete old data immediately?

Reply:
Seems in `insertOverwrite` process, the stale data can be removed immediately.  I think we can keep the original `addToStaleFolders` function here

> Let's discuss insertOverwrite behavior.
> Should we delete old data immediately?

Seems in `insertOverwrite` process, the stale data can be removed immediately.  I think we can keep the original `addToStaleFolders` function here

Reply:
> Let's discuss insertOverwrite behavior.
> Should we delete old data immediately?

Seems in `insertOverwrite` process, the stale data can be removed immediately.  I think we can keep the original `addToStaleFolders` function here

Reply:
Seems in `insertOverwrite` process, the stale data can be removed immediately.  I think we can keep the original `addToStaleFolders` function here